engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/ModCheckValidator.java

```diff
@@ -24,7 +22,8 @@
  * @deprecated As of release 5.1.0, replaced by {@link Mod10CheckValidator} and {@link Mod11CheckValidator}
  */
 @Deprecated
-public class ModCheckValidator extends ModCheckBase implements ConstraintValidator<ModCheck, CharSequence> {
+@SuppressWarnings("deprecation")
+public class ModCheckValidator extends ModCheckBase implements ConstraintValidator<org.hibernate.validator.constraints.ModCheck, CharSequence> {
```

Comment:
Isn't the fact that the class is marked as deprecated sufficient?

I don't like to use the FQCN as a workaround. It's not pretty at all.

Reply:
Just a quick answer on this one - the compiler was still not happy about the imports of deprecated stuff and as a workaround I've used the fqcn. But let me double check that as I might be mixing the warnings from the IDEA with the ones from the build.